src/ng/directive/ngInclude.js

```diff
@@ -253,6 +255,8 @@ var ngIncludeDirective = ['$templateRequest', '$anchorScroll', '$animate',
               currentScope.$emit('$includeContentLoaded', src);
               scope.$eval(onloadExp);
             }, function() {
+              if (scope.$$destroyed) return;
```

Comment:
I do not think there is a need for a lot of spying, you can have another test just like the one you have but
* same template but without the `ng-if`
* creating a new scope from `$rootScope`, linking the template to it
* registering a listener
* destroying the scope
* doing the http flush